src/rgw/rgw_zone.h

```diff
@@ -975,9 +975,25 @@ struct RGWPeriodLatestEpochInfo {
 };
 WRITE_CLASS_ENCODER(RGWPeriodLatestEpochInfo)
 
+
+/*
+ * RGWPeriod represents a period of time during which a single
+ * configuration of multisite was used.
+ *
+ * Although not represented in the RGWPeriod class, this configuration
+ * 0 or 1 realms, 1 or more zone groups, and for each zone group, 1 or
+ * more zones. And the configuration has addtional information as
+ * well.
+ *
+ * When the configuration is changed, a new RGWPeriod is created with
+ * a new id (uuid). And a new RGWPeriod is linked back to its
+ * predecessor RGWPeriod through the predecessor_uuid field, thus
+ * creating a "linked list"-like structure of RGWPeriods back to the
+ * cluster's creation.
+ */
```

Comment:
the RGWPeriod object contains the entire configuration of a RGWRealm, including its RGWZoneGroups and RGWZones. consistency of this configuration is maintained across all zones by passing around the period object in its json representation

if a new configuration changes which zone is the metadata master zone (ie master zone of the master zonegroup), then a new period id is generated, its realm_epoch is incremented, and the RGWRealm object is updated to reflect that new current_period id and epoch. if the configuration does not change which zone is the metadata master, then only the RGWPeriod::epoch is incremented

so i guess you would say that it represents the period of time that a given zone was in charge - or put differently, the period of time between failover events

Reply:
Thanks, @cbodley! I updated the comment to include some of your text. If you wouldn't mind looking the new version over at a time of your convenience, I would appreciate it.